app/controllers/application_controller.rb

```diff
@@ -1398,10 +1398,18 @@ def set_config(db_record) # pass in the db record, either @host or @vm
       # Add disks to the device array
       unless db_record.hardware.disks.nil?
         db_record.hardware.disks.each do |disk|
-          loc = disk.location.nil? ? "" : disk.location
-          dev = disk.controller_type ? disk.controller_type << " " << loc : ""  # default device is controller_type
-          desc = disk.filename                              # default description is filename
-          icon = disk.device_name                       # default icon prefix is device_name
+          # relying on to_s to force nils into ""
+          loc = disk.location.to_s
+
+          # default device is controller_type
+          dev = disk.controller_type ? (disk.controller_type << " " << loc) : ""
```

Comment:
Fixed, thanks :)

the rest of the function suffers the same problem, but that should be fixed in [#7175](https://github.com/ManageIQ/manageiq/pull/7175/files#diff-55c5b7aecfb519d0e4880eaf2788eb6eL1402) :)